src/renderers/rust/templates/instructionsCpiPageBuilder.njk

```diff
@@ -0,0 +1,67 @@
+/// `{{ instruction.name | snakeCase }}` CPI instruction builder.
+pub struct {{ instruction.name | pascalCase }}Builder<'a> {
+  program: &'a solana_program::account_info::AccountInfo<'a>,
+{% for account in instruction.accounts %}
+  {% if account.isSigner === 'either' %}
+    {{ account.name | snakeCase }}: Option<(&'a solana_program::account_info::AccountInfo<'a>, bool)>,
+  {% else %}
+    {{ account.name | snakeCase }}: Option<&'a solana_program::account_info::AccountInfo<'a>>,
+  {% endif %}
+{% endfor %}
+{% for arg in instructionArgs %}
+  {% if not arg.default %}
+    {{ arg.name | snakeCase }}: {{ arg.type if arg.innerOptionType else 'Option<' + arg.type + '>' }},
+  {% endif %}
+{% endfor %}
+}
+
+impl<'a> {{ instruction.name | pascalCase }}Builder<'a> {
+  pub fn new(program: &'a solana_program::account_info::AccountInfo<'a>) -> Self {
+    Self {
+      program,
+      {% for account in instruction.accounts %}
+        {{ account.name | snakeCase }}: None,
+      {% endfor %}
+      {% for arg in instructionArgs %}
+        {% if not arg.default %}
+          {{ arg.name | snakeCase }}: None,
+        {% endif %}
+      {% endfor %}
+    }
+  }
+  {% for account in instruction.accounts %}
+    pub fn {{ account.name | snakeCase }}(&'a mut self, {{ account.name | snakeCase }}: &'a solana_program::account_info::AccountInfo<'a>{{ ', as_signer: bool' if account.isSigner === 'either' }}) -> &mut Self {
```

Comment:
```suggestion
    pub fn {{ account.name | snakeCase }}(&mut self, {{ account.name | snakeCase }}: &'a solana_program::account_info::AccountInfo<'a>{{ ', as_signer: bool' if account.isSigner === 'either' }}) -> &mut Self {
```

The reference to self shouldn't have the same lifetime as the `AccountInfo` reference and its struct fields' references, otherwise this build error occurs:

```
error[E0597]: `builder` does not live long enough
  --> onchain-consumer/src/lib.rs:11:5
   |
8  | pub fn etp<'a>(program_id: &Pubkey, accounts: &'a [AccountInfo<'a>], instruction_data: &[u8]) -> ProgramResult {
   |            -- lifetime `'a` defined here
9  |     let mut builder = AddConfigLinesBuilder::new(&accounts[0]);
   |         ----------- binding `builder` declared here
10 |     let candy_machine = &accounts[1];
11 |     builder.candy_machine(candy_machine);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     borrowed value does not live long enough
   |     argument requires that `builder` is borrowed for `'a`
12 |     Ok(())
13 | }
   | - `builder` dropped here while still borrowed
```